presto-hive/src/main/java/com/facebook/presto/hive/HiveCompressionCodec.java

```diff
@@ -20,24 +20,44 @@
 import parquet.hadoop.metadata.CompressionCodecName;
 
 import java.util.Optional;
+import java.util.function.Predicate;
 
+import static com.facebook.presto.hive.HiveStorageFormat.DWRF;
+import static com.facebook.presto.hive.HiveStorageFormat.ORC;
 import static java.util.Objects.requireNonNull;
 
 public enum HiveCompressionCodec
 {
-    NONE(null, CompressionKind.NONE, CompressionCodecName.UNCOMPRESSED),
-    SNAPPY(SnappyCodec.class, CompressionKind.SNAPPY, CompressionCodecName.SNAPPY),
-    GZIP(GzipCodec.class, CompressionKind.ZLIB, CompressionCodecName.GZIP);
+    NONE(null, CompressionKind.NONE, CompressionCodecName.UNCOMPRESSED, allStorageFormatsSupported().negate()),
+    SNAPPY(SnappyCodec.class, CompressionKind.SNAPPY, CompressionCodecName.SNAPPY, allStorageFormatsSupported()),
+    GZIP(GzipCodec.class, CompressionKind.ZLIB, CompressionCodecName.GZIP, allStorageFormatsSupported()),
+    ZSTD(null, CompressionKind.ZSTD, null, supportedStorageFormatsForZSTD());
 
     private final Optional<Class<? extends CompressionCodec>> codec;
     private final CompressionKind orcCompressionKind;
-    private final CompressionCodecName parquetCompressionCodec;
+    private final Optional<CompressionCodecName> parquetCompressionCodec;
+    private final Predicate<HiveStorageFormat> supportedStorageFormats;
 
-    HiveCompressionCodec(Class<? extends CompressionCodec> codec, CompressionKind orcCompressionKind, CompressionCodecName parquetCompressionCodec)
+    HiveCompressionCodec(
+            Class<? extends CompressionCodec> codec,
+            CompressionKind orcCompressionKind,
+            CompressionCodecName parquetCompressionCodec,
+            Predicate<HiveStorageFormat> supportedStorageFormats)
     {
         this.codec = Optional.ofNullable(codec);
         this.orcCompressionKind = requireNonNull(orcCompressionKind, "orcCompressionKind is null");
-        this.parquetCompressionCodec = requireNonNull(parquetCompressionCodec, "parquetCompressionCodec is null");
+        this.parquetCompressionCodec = Optional.ofNullable(parquetCompressionCodec);
+        this.supportedStorageFormats = requireNonNull(supportedStorageFormats);
```

Comment:
@jainxrohit We tend to add a message like this: 

`requireNonNull(supportedStorageFormats, "supportedStorageFormats is null")`